swiper.el

```diff
@@ -733,53 +733,36 @@ When capture groups are present in the input, print them instead of lines."
   (when (bound-and-true-p evil-mode)
     (evil-set-jump)))
 
-(declare-function char-fold-to-regexp "char-fold")
+(defun swiper--normalize-regex (re)
+  "Normalize the swiper regex RE.
+Add a space after a leading `^' if needed and apply
+`search-default-mode' if bound."
+  (replace-regexp-in-string
+   "^\\(\\(?:\\\\(\\)?\\^\\)"
+   (concat "\\1" (if (eq 'swiper (ivy-state-caller ivy-last)) " " ""))
+   (if (fboundp (bound-and-true-p search-default-mode))
+       (mapconcat
+        (lambda (x)
+          (if (string-match-p "^[^$\\^]+$" x)
+              (funcall search-default-mode x)
+            x))
+        (split-string re "\\b" t) "")
+     re)))
```

Comment:
You should probably pass a non-`nil` `FIXEDCASE` argument to `replace-regexp-in-string`, right?

Reply:
I don't understand why we would need that. The substituted text is just the original text with an optional space at the end.

Reply:
Sure, but in library code I think `FIXEDCASE` and `LITERAL` should always default to non-`nil` unless these features are needed. It's both more efficient and less surprising that way. We don't want Emacs trying to guess whether to change the case of substitution text here.